compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -364,17 +539,95 @@ public void performCodeGen(ModuleContext moduleContext, CompilationCache compila
             return;
         }
         boolean isRemoteMgtEnabled = moduleContext.project().buildOptions().compilationOptions().remoteManagement();
-        CompiledJarFile compiledJarFile = jvmCodeGenerator.generate(bLangPackage, isRemoteMgtEnabled);
+        CompiledJarFile compiledJarFile = jvmCodeGenerator.generate(bLangPackage, false, isRemoteMgtEnabled);
+        if (compiledJarFile == null) {
+            throw new IllegalStateException("Missing generated jar, module: " + moduleContext.moduleName());
+        }
+        String jarFileName = getJarFileName(moduleContext) + JAR_FILE_NAME_SUFFIX;
+        try {
+            ByteArrayOutputStream byteStream = JarWriter.write(compiledJarFile, getResources(moduleContext));
+            compilationCache.cachePlatformSpecificLibrary(this, jarFileName, byteStream, false);
+        } catch (IOException e) {
+            throw new ProjectException("Failed to cache generated jar, module: " + moduleContext.moduleName());
+        }
+        // skip generation of the test jar if --with-tests option is not provided
+        if (moduleContext.project().buildOptions().skipTests()) {
+            return;
+        }
+
+        if (!bLangPackage.hasTestablePackage()) {
+            return;
+        }
```

Comment:
Combine these two into one condition

Reply:
Fixed in aa8254a61e2f3d88c1fdd944ffe5a5b257ede583.